R/kernel.r

```diff
@@ -169,94 +192,100 @@ complete = function(request) {
         
     # .guessTokenFromLine, like most other functions here usually sets variables in .CompletionEnv.
     # When specifying update = FALSE, it instead returns a list(token = ..., start = ...)
-    c.info <- c(list(comps = utils:::.retrieveCompletions()),
-                utils:::.guessTokenFromLine(update = FALSE))
+    c.info <- c(
+        list(comps = utils:::.retrieveCompletions()),
+        utils:::.guessTokenFromLine(update = FALSE))
     
     # good coding style for completions
     comps <- gsub('=$', ' = ', c.info$comps)
     
-    start_position = chars_before_line + c.info$start
+    start_position <- chars_before_line + c.info$start
     send_response('complete_reply', request, 'shell', list(
         matches = as.list(comps),  # make single strings not explode into letters
         metadata = namedlist(),
         status = 'ok',
         cursor_start = start_position,
-        cursor_end = start_position + nchar(c.info$token)
-    ))
+        cursor_end = start_position + nchar(c.info$token)))
 },
 
 history = function(request) {
-  send_response("history_reply", request, 'shell', list(history=list()))
+    send_response('history_reply', request, 'shell', list(history = list()))
 },
 
 kernel_info = function(request) {
-  rversion = paste(version$major, '.', version$minor, sep="")
-  send_response("kernel_info_reply", request, 'shell',
-                list(protocol_version='5.0', implementation='IRkernel',
-                     implementation_version='0.2',
-                     language_info=list(name="R", codemirror_mode="r",
-                        pygments_lexer="r", mimetype="text/x-r-source",
-                        file_extension=".r", version=rversion
-                     ),
-                     banner=version$version.string
-                    )
-                )
+    rversion <- paste0(version$major, '.', version$minor)
+    send_response('kernel_info_reply', request, 'shell', list(
+        protocol_version       = '5.0',
+        implementation         = 'IRkernel',
+        implementation_version = '0.2',
```

Comment:
We should probably define a global version variable, so that it's easier to remember to update it.